backend/src/main/java/com/example/backend/service/RecipeService.java

```diff
@@ -19,12 +19,10 @@ public List<Recipe> getAllRecipes() {
     }
 
     public Recipe addRecipe(Recipe recipe) {
-        Recipe recipeToSave = recipe.withId(recipe.id());
-        return recipeRepository.save(recipeToSave);
+        return recipeRepository.save(recipe);
     }
 
-    public Recipe getRecipeById(String id) {
-        return recipeRepository.findById(id)
-                .orElseThrow(NoSuchElementException::new);
+    public Optional<Recipe> getRecipeById(String id) {
+        return recipeRepository.findById(id);
     }
```

Comment:
the way you used before was much better so your test could also test the exception